backend/app/moderation.py

```diff
@@ -0,0 +1,312 @@
+                request_data=json.dumps(
+                    {"keys": keys, "current_values": current_values}
+                ),
+                is_new_submission=is_new_submission,
+                is_outdated=False,
+                build_id=appdata.build_id,
+                job_id=appdata.job_id,
+            )
+            new_requests.append(request)
+
+    if len(new_requests) > 0:
+        # Mark previous requests as outdated, to avoid flooding the moderation queue with requests that probably aren't
+        # relevant anymore. Outdated requests can still be viewed and approved, but they're hidden by default.
+        sqldb.session.query(models.ModerationRequest).filter_by(
+            appid=request.appid
+        ).update({"is_outdated": True})
+
+        for request in new_requests:
+            sqldb.session.add(request)
+        sqldb.session.commit()
+
+        return ReviewRequestResponse(
+            requires_review=not config.settings.moderation_observe_only
+        )
+    else:
+        return ReviewRequestResponse(requires_review=False)
+
+
+class Review(BaseModel):
+    approve: bool
+    comment: str | None
+
+    @validator("comment", always=True)
+    def reject_requires_comment(cls, v, values):
+        if v is None and not values["approve"]:
+            raise ValueError("rejecting a request requires a comment")
+        return v
+
+
+@router.post("/requests/{id}/review", status_code=204)
+def submit_review(
+    id: int,
+    review: Review,
+    login=Depends(logins.login_state),
+    _moderator=Depends(moderator_only),
+):
+    """Approve or reject the moderation request with a comment. If all requests for a job are approved, the job is
+    marked as successful in flat-manager."""
+
+    request = (
+        sqldb.session.query(models.ModerationRequest)
+        .filter_by(id=id)
+        .with_for_update()
+        .first()
+    )
+
+    if request is None:
+        raise HTTPException(status_code=404, detail="not_found")
+    elif request.handled_at is not None:
+        raise HTTPException(status_code=400, detail="already_handled")
+
+    request.is_approved = review.approve
+    request.handled_by = login["user"].id
+    request.handled_at = func.now()
+    request.comment = review.comment
+
+    sqldb.session.commit()
+
+    if request.is_approved:
+        # Check if all requests for the job are approved
+        remaining = (
+            sqldb.session.query(models.ModerationRequest)
+            .filter_by(job_id=request.job_id)
+            .filter(models.ModerationRequest.is_approved is not True)
+            .count()
+        )
+        if remaining == 0:
+            # Tell flat-manager that the job is approved
+            worker.review_check.send(request.job_id, "Passed", None)
+    else:
+        # If any request is rejected, the job is rejected
+        worker.review_check.send(
+            request.job_id, "Failed", "The review was rejected by a moderator."
```

Comment:
I'm not a native speaker, but it seems a bit weird to me, that the `review` was rejected? Wouldn't it be something else, like the upload/build/job?